### PR TITLE
Dev9

### PR DESCRIPTION
**Pull request by AndreiBakunovich**
WARNING! СМОТРЕТЬ ССЫЛКУ https://github.com/AndreiBakunovich/TAT-tasks/pull/1/files

### CODE REVIEW

File: dev9/Dev9/Program.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Dev9
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string[] lines = new string[2];
+            string path = @"c:\Users\source.txt";
+            try {
```
Comment (IgorKheidorov) [inline dev9/Dev9/Program.cs]: Code style, this is C#, use 
method()
{
}
instead of
method(){
}

File: dev9/Dev9/Program.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using System.IO;
+
+namespace Dev9
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string[] lines = new string[2];
+            string path = @"c:\Users\source.txt";
```
Comment (IgorKheidorov) [inline dev9/Dev9/Program.cs:15]: hard code

File: dev9/Dev9/Program.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            char[] lineOneInChar = new char[lines[0].Length];
+            char[] lineTwoInChar = new char[lines[1].Length];
+            lineOneInChar = lines[0].ToCharArray();
+            lineTwoInChar = lines[1].ToCharArray();
+
+            string result;
+            Replacer rep = new Replacer();
+            //random string selection
+            Random rnd = new Random();
+            if (rnd.Next(1, 3) == 1)
```
Comment (IgorKheidorov) [inline dev9/Dev9/Program.cs]: code style

File: dev9/Dev9/Replacer.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            //dividion lines on parts
+            Random rnd = new Random();
+
+            int donorIntervalStart = rnd.Next(0, donor.Length),
+                donorIntervalFinish = rnd.Next(donorIntervalStart, donor.Length),
+                recipientIntervalStart = rnd.Next(0, recipient.Length),
+                recipientIntervalFinish = rnd.Next(recipientIntervalStart, recipient.Length);
+            
+            //result configuring
+            for (int i = 0; i < recipientIntervalStart; i++)
```
Comment (IgorKheidorov) [inline dev9/Dev9/Replacer.cs]: bad idea to copy arrays

File: dev9/Dev9/Replacer.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                res = res + recipient[i];
+
```
Comment (IgorKheidorov) [inline dev9/Dev9/Replacer.cs]: bad idea to copy arrays

File: dev9/Dev9/Replacer.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            for (int i = donorIntervalStart; i < donorIntervalFinish; i++)
+                res = res + donor[i];
+
```
Comment (IgorKheidorov) [inline dev9/Dev9/Replacer.cs]: bad idea to copy arrays

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Code style for C#, hard code

File: dev9/Dev9/Program.cs (hunk 2 above)
Comment (AndreiBakunovich) [reply dev9/Dev9/Program.cs:15]: What did you mean exactly in this code?

File: dev9/Dev9/Replacer.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dev9
+{
+    class Replacer
```
Comment (IgorKheidorov) [inline dev9/Dev9/Replacer.cs:9]: Better name StrinReplacer

File: dev9/Dev9/Program.cs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            catch
+            {
+                Console.WriteLine("Incorrect file data format.");
+                Console.ReadLine();
+                return;
+            }
+            
+            char[] lineOneInChar = new char[lines[0].Length];
+            char[] lineTwoInChar = new char[lines[1].Length];
+            lineOneInChar = lines[0].ToCharArray();
```
Comment (IgorKheidorov) [inline dev9/Dev9/Program.cs:35]: What for? you make too many unnecessary convertions from char to string and back, not effective

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**